client/index.ts

```diff
@@ -211,19 +212,19 @@ export function SecretAgentClientGenerator(
 
     /////// METHODS THAT DELEGATE TO ACTIVE TAB //////////////////////////////////////////////////////////////////////////
 
-    public async goto(href: string) {
+    public goto(href: string) {
```

Comment:
Why remove `async`? I get that it's not "needed" but it's kinda nice for the code to signify what it returns. Should we give it an explicit return type?

Reply:
Async without the await makes this kill off any further upstream stack trace. This is what I was telling you about that was behaving very strangely. I personally don't like cluttering the code with return types, but I can definitely see the argument for it, and maybe I'm just ok with it because the IDE is figuring it out from the ts compiler. I think if we decide to do that, we should probably make a global eslint rule requiring return types.

Reply:
I added a rule to *-client, *-interfaces and commons to require return types on functions. I think this is a better direction for our code, but don't want to do it for all the code in one shot.

Reply:
Great